packages/eth-edge/src/services/alchemy_node_facade.service.ts

```diff
@@ -51,21 +52,34 @@ export class AlchemyNodeServiceFacade {
         // from ETHER to WEI so we can represent them as BigIntegers
         return Promise.all(
             transfers.map(async (transfer) => {
-                const transaction = await this.alchemy.getTransaction(transfer.hash);
-                return {
+                const transaction: Transaction = {
                     id: -1,
                     transactionAddress: transfer.hash,
                     walletIn: transfer.to,
-                    contactIn: null,
                     walletOut: transfer.from,
-                    contactOut: null,
                     amount: Web3.utils.toWei(normalizeCurrency(transfer.value), "ether"),
-                    gasPrice: transaction.gasPrice.toString(),
-                    gasLimit: transaction.gasLimit.toString(),
                     createdAt: new Date(transfer.metadata.blockTimestamp),
                     tags: [],
                 };
+
+                const transactionDetails = await this.getTransaction(transfer.hash);
+                if (transactionDetails) {
+                    transaction.gasPrice = transactionDetails.gasPrice.toString();
+                    transaction.gasLimit = transactionDetails.gasLimit.toString();
+                }
+
+                return transaction;
             }),
         );
     }
+
+    private async getTransaction(hash: string): Promise<TransactionResponse> {
+        // This function can fail sometimes from the ethereum node side so in that case we just won't assign the
+        // transaction the extra details, in the future we can try to implement a retry mechanism on this
+        try {
+            return this.alchemy.getTransaction(hash);
```

Comment:
This should be awaited before the return so if the promise is rejected through an exception the exception will be handled in this try catch instead of downstream

```suggestion
            return await this.alchemy.getTransaction(hash);
```